mlrun/api/main.py

```diff
@@ -224,6 +225,85 @@ async def move_api_to_online():
         if get_k8s_helper(silent=True).is_running_inside_kubernetes_cluster():
             _start_periodic_cleanup()
             _start_periodic_runs_monitoring()
+            _start_logs_collection()
+
+
+def _start_logs_collection():
+    if config.sidecar.logs_collector.mode == mlrun.api.schemas.LogsCollectorMode.legacy:
+        logger.info(
+            "Using legacy logs collection method, skipping logs collection periodic function",
+            mode=config.sidecar.logs_collector.mode,
+        )
+        return
+    logger.info(
+        "Starting logs collection periodic function",
+        mode=config.sidecar.logs_collector.mode,
+        interval=config.sidecar.logs_collector.interval,
+    )
+    run_function_periodically(
+        config.sidecar.logs_collector.interval,
+        _collect_runs_logs.__name__,
+        False,
+        _collect_runs_logs,
+    )
+
+
+async def _collect_runs_logs():
+    logs_collector_client = mlrun_log_collector.LogCollectorClient()
+    db_session = create_session()
+    try:
+        runs_to_update_requested_logs = []
+        # list all the runs in the system which we didn't request logs collection for yet
+        runs = await fastapi.concurrency.run_in_threadpool(
+            get_db().list_distinct_runs_uids,
+            db_session,
+            project="*",
+            requested_logs=False,
+            only_uids=False,
+        )
+        for run in runs:
```

Comment:
Wrap below in a function, use asyncio gather, chunked to 5-10 at a time. If few fails - retry until list is empty or few failed for x times (let's say 3). Then failed all.